app/src/main/java/com/sun/cookbook/data/source/RecipeRepository.kt

```diff
@@ -0,0 +1,26 @@
+package com.sun.cookbook.data.source
+
+import com.sun.cookbook.data.model.RecipeSlide
+import com.sun.cookbook.data.source.local.RecipeLocalDataSource
+import com.sun.cookbook.data.source.remote.OnFetchDataJsonListener
+import com.sun.cookbook.data.source.remote.RecipeRemoteDataSource
+
+class RecipeRepository private constructor(
+    private val remote: RecipeDataSource.Remote,
+    private val local: RecipeDataSource.Local
+) {
+    private object Holder {
+        val INSTANCE = RecipeRepository(
+            remote = RecipeRemoteDataSource.instance,
+            local = RecipeLocalDataSource.instance
+        )
+    }
+
+    fun getRecipeSlide(listener: OnFetchDataJsonListener<MutableList<RecipeSlide>>) {
+        remote.getDataRecipeSlide(listener)
+    }
+
+    companion object {
+        val instance: RecipeRepository by lazy { Holder.INSTANCE }
+    }
+}
```

Comment:
Mình suggest bạn viết class singleton như này nhé 
```suggestion
class RecipeRepository private constructor(
    private val remote: RecipeDataSource.Remote,
    private val local: RecipeDataSource.Local
) {
  
 companion object {
@Volatile private var INSTANCE: Singleton ? = null
 fun  getInstance(): Singleton {
     INSTANCE ?:    synchronized(this) {
         INSTANCE ?: Singleton(param).also { INSTANCE = it }
     }
        
}

}
```